Make required folder structure more flexible

enryH opened the pull request:
- [x] allow components in all folders currently in examples

- [x] streamlit
  - [x] main folder
  - [x] section folder
  - [x] subsection folder

- [x] quarto reports
  - [x] main folder
  - [x] section folder
  - [x] subsection folder

enryH commented:
Next would be to go even deeper and collect subfolder components to the subsection reports. This should be possible only in `config_manager.py` and be independent of the specific report. Or maybe even better to generate additional subsections which are just appended. To be decided in a separate PR.

sayalaruano commented on src/vuegen/streamlit_reportview.py:
```diff
@@ -114,6 +149,18 @@ def generate_report(
+                    section.file_path = subsection_file_path
+                    # Create a Page object for each subsection and add it to the home page content
+                    report_manag_content.append(
+                        f"{section_name_var}_overview = st.Page('{subsection_file_path}', title='Overview {section.title}')"
```
Maybe the title can be just Overview to avoid repetition bc it's always under the `section.title`

enryH replied:
I think the `title='Overview {section.title}'` is even ignored. The subsection need unique names. So you cannot have just overview anywhere. I think the title is set based on the variable name.

sayalaruano reviewed (changes requested):
In general, it works as expected, but there are some unintended behaviours: 

-  In a streamlit report, if there is a `description.md` file for the report and there are components in the main folder, the description is shown twice.
- In the streamlit report, instead of putting all the components from the main folder in the Home page, would not be better to create an ´Overview´ section, and keep the home section just for the title, description, and a graphical abstract if available?
-  The revelajs' report structure is broken, I attached screenshots with the stable version and the one from this branch. Each subsection should be a tab in the section.

**Stable version**
![flexible_folder_str_version](https://github.com/user-attachments/assets/2d6b4108-7151-4309-82c4-9631cc03e385)

**Flexible folder str version**
![stable_version](https://github.com/user-attachments/assets/43e370f3-8e24-4552-8472-3930956879f9)

enryH commented:
I think I will need to add `panel-tabset` at some places. This lets me wonder: should we have more than qmd file as output? the qmd files change slightly depending on the output report, so maybe having only one qmd might be misleading one to think (like me) that all are alike.

sayalaruano commented:
I used the basic case study folder and created my own example; in both cases, the revealjs structure was not good. So, what would be the alternative to create just one qmd file?

enryH commented:
Ok. It was a string interpolation issue. I removed a f-string which removed one level of the curly brackets before... Should be working now. I added tabs to the overview sections.

enryH commented:
Before we add this, I would still adapt the basic example to include some components on the section folders and maybe even root of the example. But for the root, we maybe should have an even simpler example

```
root
|-- folder
|-  | - description.md
|--description.md
```

enryH commented:
I noted done the additional testing requirments for a future PR in #116
